cetmix_tower_server_queue/models/cx_tower_file.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CxTowerFile(models.Model):
+    _inherit = "cx.tower.file"
+
+    is_processed = fields.Boolean(
+        string="Is Processed",
+        default=False,
+        help="True if the file is already processed",
+    )
+
+    def upload(self, raise_error=False):
+        """
+        Trigger asynchronous upload via job queue.
+        """
+        self.with_delay()._do_upload(raise_error=raise_error)
+
+    def download(self, raise_error=False):
+        """
+        Trigger asynchronous download via job queue.
+        """
+        self.with_delay()._do_download(raise_error=raise_error)
+
+    def _do_upload(self, raise_error=False):
+        """
+        Perform the upload and update processing status.
+        Sets 'is_processed' to True before the upload and to False after.
+        """
+        self.ensure_one()
+        self.is_processed = True
```

Comment:
We should set this flag outside this function because this method is queued. So we must ensure that no other task will be queued while this one is waiting.
Also this flag simply will not work because it's set and reset within the same transaction.